src/Microsoft.Data/DataFrame.IO.cs

```diff
@@ -0,0 +1,261 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Data
+{
+
+    public partial class DataFrame
+    {
+        private static Type GuessKind(int col, List<string[]> read)
+        {
+            Type res = typeof(string);
+            int nbline = 0;
+            foreach (var line in read)
+            {
+                if (col >= line.Length)
+                    throw new FormatException(string.Format(Strings.LessColumnsThatExpected, nbline + 1)); // Will this really work?
+
+                var val = line[col];
+                try
+                {
+                    bool.Parse(val);
+                    res = DetermineType(nbline == 0, typeof(bool), res);
+                    ++nbline;
+                    continue;
+                }
+                catch (Exception)
+                {
+                    if (string.IsNullOrEmpty(val))
+                    {
+                        res = DetermineType(nbline == 0, typeof(bool), res);
+                        continue;
+                    }
+                }
+                try
+                {
+                    float.Parse(val);
+                    res = DetermineType(nbline == 0, typeof(float), res);
+                    ++nbline;
+                    continue;
+                }
+                catch (Exception)
+                {
+                    if (string.IsNullOrEmpty(val))
+                    {
+                        res = DetermineType(nbline == 0, typeof(float), res);
+                        continue;
+                    }
+                }
+                res = DetermineType(nbline == 0, typeof(string), res);
+                ++nbline;
+            }
+            return res;
+        }
+
+        private static Type DetermineType(bool first, Type suggested, Type previous)
+        {
+            if (first)
+                return suggested;
+            else
+                return MaxKind(suggested, previous);
+        }
+
+        private static Type MaxKind(Type a, Type b)
+        {
+            if (a == typeof(string) || b == typeof(string))
+                return typeof(string);
+            if (a == typeof(float) || b == typeof(float))
```

Comment:
What about `double` or `int`, etc?

Reply:
Right, `MaxKind` only has 1 caller `DetermineType` and `DetermineKind` always passes in one of string, float or bool. Or do you mean to add support now to make it easier to parse `double`/`int` in the future?

Reply:
Maybe the question is - why aren't we supporting `double` and `int` types? Why just `float`?

Reply:
Ah, that's because we only read in the first 10 lines in the csv file and try to guess the types from it. Since the first 10 lines of column A could be `int`, but everything else after that a `float`, I chose `float` as the common numeric type. At some point, it'd probably be worth it to have a strongly typed API that specifies the column types